src/Streamly/Internal/Data/Stream/Zip/Concurrent.hs

```diff
@@ -0,0 +1,47 @@
+{-# LANGUAGE UndecidableInstances #-}
+-- |
+-- Module      : Streamly.Internal.Data.Stream.Concurrent
+-- Copyright   : (c) 2017 Composewell Technologies
+-- License     : BSD-3-Clause
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+--
+-- To run examples in this module:
+--
+-- >>> import qualified Streamly.Prelude as Stream
+--
+module Streamly.Internal.Data.Stream.Zip.Concurrent
+    (
+      ZipConcurrent (..)
+    )
+where
+
+import Streamly.Internal.Data.Stream (Stream)
+import Streamly.Internal.Data.Stream.Concurrent (MonadAsync, zipWith)
+
+import qualified Streamly.Internal.Data.Stream as Stream (repeat)
+import Prelude hiding (map, repeat, zipWith)
+
+-- $setup
+-- >>> import qualified Streamly.Internal.Data.Stream as Stream
+-- >>> import qualified Streamly.Data.Fold as Fold
+--
+
+newtype ZipConcurrent m a = ZipConcurrent {getStream :: Stream m a}
+        deriving (Semigroup, Monoid, Functor)
```

Comment:
Rather than using the Semigroup/Monoid of the Serial stream we may want to use concurrent versions of those. Remove these two instances.